### PR TITLE
Improve the floating shell button and fix its position in the screens

### LINKED ISSUE #16887: Improve floating shell button
### Summary

Improve floating shell button and fix spacing as mentioned in https://www.figma.com/file/Tf5nfkYvpbnNCo4rKLK7lS?node-id=3674:338290&mode=design#511287891

### PR DESCRIPTION
**Pull request by Parveshdhull**
fixes https://github.com/status-im/status-mobile/issues/16887

### Testing request:
Please check jump-to button height in all screens (home, communities, messages etc.)

##### Known issue:
https://github.com/status-im/status-mobile/issues/16331#issuecomment-1623597644

status: ready

### CODE REVIEW

File: src/quo2/components/navigation/floating_shell_button/style.cljs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  (reanimated/apply-animations-to-style
+   (if opacity-anim
+     {:opacity opacity-anim}
+     {})
+   (merge
+    {:flex-direction     :row
+     :height             44
+     :align-self         :center
+     :padding-top        8
+     :padding-horizontal 12
```
Comment (Parveshdhull) [inline src/quo2/components/navigation/floating_shell_button/style.cljs:15]: Except file rename and break into view and style, only this section changed in floating-shell-button component

File: src/status_im2/contexts/chat/composer/view.cljs [hunk 2]
```diff
@@ -66,62 +66,64 @@
     (effects/link-previews props state animations subs)
     (effects/images props state animations subs)
     [:<>
-     [sub-view/shell-button state animations subs]
```
Comment (Parveshdhull) [inline src/status_im2/contexts/chat/composer/view.cljs:69]: shell button moved to sub component

File: src/status_im2/contexts/chat/composer/view.cljs [hunk 3]
```diff
@@ -66,62 +66,64 @@
...
-           :max-font-size-multiplier 1
-           :multiline                true
-           :placeholder              (i18n/label :t/type-something)
-           :placeholder-text-color   (colors/theme-colors colors/neutral-30 colors/neutral-50)
-           :style                    (style/input-text props state subs max-height)
-           :max-length               constants/max-text-size
-           :accessibility-label      :chat-message-input}]]
+     [rn/view
+      {:style style/composer-sheet-and-jump-to-container}
+      [sub-view/shell-button state]
```
Comment (Parveshdhull) [inline src/status_im2/contexts/chat/composer/view.cljs:77]: new-changes (L75-77)

File: src/quo2/components/navigation/floating_shell_button/view.cljs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+     {:position :absolute
+      :right    8}]]
+   ;; Mid Section (jump-to)
+   [dynamic-button-view :jump-to dynamic-buttons nil]
+   ;; Right Section
+   [section
+    [rn/view
+     {:style {:position       :absolute
+              :flex-direction :row
+              :right          0}}
```
Comment (OmarBasem) [inline src/quo2/components/navigation/floating_shell_button/view.cljs]: style file

File: src/status_im2/contexts/chat/composer/sub_view.cljs [hunk 5]
```diff
@@ -27,12 +25,8 @@
   [:f> f-blur-view layout-height focused?])
 
 (defn- f-shell-button
-  [{:keys [maximized? focused?]} {:keys [height]} {:keys [reply edit]}]
+  [{:keys [focused?]}]
   (let [customization-color (rf/sub [:profile/customization-color])
-        insets              (safe-area/get-insets)
-        extra-height        (utils/calc-top-content-height reply edit)
-        neg-y               (utils/calc-shell-neg-y insets maximized? extra-height)
-        y-container         (reanimated/use-shared-value neg-y)
```
Comment (OmarBasem) [inline src/status_im2/contexts/chat/composer/sub_view.cljs:35]: Checking if reply or edit not needed?

**Review by OmarBasem — APPROVED** [review]

File: src/quo2/components/navigation/floating_shell_button/view.cljs (hunk 4 above)
Comment (Parveshdhull) [reply src/quo2/components/navigation/floating_shell_button/view.cljs]: fixed, thank you

File: src/status_im2/contexts/chat/composer/sub_view.cljs (hunk 5 above)
Comment (Parveshdhull) [reply src/status_im2/contexts/chat/composer/sub_view.cljs:35]: As jump-to button is now part of sheet itself, it gets pushed up according to the size of sheet. We don't need to manually calculate position anymore.

**Comment by churik** [comment]
@Parveshdhull 
the issue with e2e is that currently on the screen there might be several jump-to buttons.

Appium 2023-08-14 15-49-40

Is there any way to distinguish somehow the one which is visible?

**Comment by VolodLytvynenko** [comment]
HI @Parveshdhull Thank you for PR. Here's a list of the issues that have been found:

## ISSUE 1: [IOS] Textbox disappears into background and keyboard can't be closed after pasting long text

### Steps to Reproduce:
1. Open the chat composer.
2. Paste a long text into the composer. (in my case I pasted 2000 chars)

### Actual Result:
The textbox containing the pasted long text disappears into the background, and the keyboard cannot be closed 

https://github.com/status-im/status-mobile/assets/52490791/eb3fa4d7-aad2-4899-8f3e-1f92966a5946


### Expected Result:
The textbox should remain visible.
How it works on the last nightly:

https://github.com/status-im/status-mobile/assets/52490791/31416fdb-8131-4044-bdbd-fc6e49dad939


### ENV:
- Real device: iPhone 11 Pro max, IOS 16
- Real device: iPhone 11 Pro, IOS 16

**Comment by VolodLytvynenko** [comment]
## ISSUE 2: [iOS] Incorrect positioning of 'Jump To' button after tapping 'Scroll Down' Button
### Steps to Reproduce:
1. Navigate to a chat where a sufficient number of messages are displayed.
2. Scroll up 
3. Tap the 'Scroll Down' button.

### Actual Result:
In some cases, the 'Jump To' button is displayed at the same level as a message after tapping the 'Scroll Down' button.

https://github.com/status-im/status-mobile/assets/52490791/4525a574-fff0-4c24-bdcf-e96e831ea7e0



### Expected Result:
After tapping the 'Scroll Down' button, the 'Jump To' button should be positioned under the message with appropriate spacing.

### ENV:
- Real device: iPhone 11 Pro max, IOS 16
- Real device: iPhone 11 Pro, IOS 16

**Comment by Parveshdhull** [comment]
hi @VolodLytvynenko, thank you very much for testing the PR.
Issue 2 looks similar to https://github.com/status-im/status-mobile/issues/16331#issuecomment-1623597644, please can you check

**Comment by VolodLytvynenko** [comment]
> hi @VolodLytvynenko, thank you very much for testing the PR.
Issue 2 looks similar to https://github.com/status-im/status-mobile/issues/16331#issuecomment-1623597644, please can you check

@Parveshdhull My apologizes. I agree with you.  During testing forgot about this existing issue

**Comment by churik** [comment]
Ok, I identified an issue:
image

If jump-to is on a floating screen, there is no accessibility-id on it anymore and Appium cannot tap on it, and tried to tap on other jump-to button (which are below the current view and inaccessible)

**Comment by Parveshdhull** [comment]
hi @VolodLytvynenko, @churik. Thank you very much for testing the PR, both issues should be fixed now.

**UPD**: By both issues I mean Issue 1 and e2e problem.

**Comment by VolodLytvynenko** [comment]
hi @Parveshdhull thank you for PR. No issues from my side
Could you check @Francesca-G this PR? Please be aware of the existing issue mentioned [here](https://github.com/status-im/status-mobile/issues/16331#issuecomment-1623597644)

**Review by Francesca-G — APPROVED** [review]
The positioning in the screens looks good to me, so I'm approving.

I'm adding the follow up required label because as you can see the button itself doesn't completely match the design.
Blue one is design, orange one is implementation: the button looks smaller and the text doesn't seem to match.

Screenshot 2023-08-21 alle 16 49 01


Feel free to solve it in a separate issue or do what works best in this case :)

**Comment by Parveshdhull** [comment]
Hi @Francesca-G, Thank you very much for testing the PR.

The component doesn't match the design because of the issue in the react-native text component.
Please check out https://github.com/status-im/status-mobile/issues/13793.

Not sure if that issue is fixable, as the issue is closed after some attempts & inactivity.
But @yqrashawn might have more information.

Also, this looks like a base issue that can also affect the design review process in other PRs, so maybe we can discuss this more in the next design meeting.

For now, I reopened the base issue. Probably we don't need a follow-up issue, as it will report the same problem. wdyt?

**Pull request merged**